include/net/aws_iot.h

```diff
@@ -187,57 +193,52 @@ typedef void (*aws_iot_evt_handler_t)(const struct aws_iot_evt *evt);
 
 /** @brief Structure for AWS IoT broker connection parameters. */
 struct aws_iot_config {
-	/** Socket for AWS IoT broker connection */
-	int socket;
-	/** Client id for AWS IoT broker connection, used when
-	 *  @kconfig{CONFIG_AWS_IOT_CLIENT_ID_APP} is set. If not set an internal
-	 *  configurable static client id is used.
+	/** AWS IoT client ID. Must be NULL terminated.
+	 *  If not set (NULL), CONFIG_AWS_IOT_CLIENT_ID_STATIC will be used.
 	 */
 	char *client_id;
-	/** Length of client_id string. */
-	size_t client_id_len;
-	/** AWS IoT endpoint host name for broker connection, used when
-	 *  @kconfig{CONFIG_AWS_IOT_BROKER_HOST_NAME_APP} is set. If not the
-	 *  static @kconfig{AWS_IOT_BROKER_HOST_NAME} is used.
+
+	/** AWS IoT broker hostname. Must be NULL terminated.
+	 *  If not set (NULL), CONFIG_AWS_IOT_BROKER_HOST_NAME will be used.
 	 */
 	char *host_name;
-	/** Length of host_name string. */
-	size_t host_name_len;
 };
 
-/** @brief Initialize the module.
+/** @brief Initialize the library.
  *
  *  @note This API must be called exactly once, and it must return
- *        successfully.
+ *	  successfully before any other library APIs can be called.
  *
- *  @param[in] config Pointer to struct containing connection parameters.
- *  @param[in] event_handler Pointer to event handler to receive AWS IoT module
- *                           events.
+ *  @param[in] event_handler Pointer to an event handler to receive library events.
  *
- *  @return 0 If successful.
- *            Otherwise, a (negative) error code is returned.
+ *  @return 0 If successful. Otherwise, a negative error code is returned
+ *	      indicating the reason for failure.
+ *  @retval -EINVAL if an invalid parameter is passed in.
  */
-int aws_iot_init(const struct aws_iot_config *const config,
-		 aws_iot_evt_handler_t event_handler);
+int aws_iot_init(aws_iot_evt_handler_t event_handler);
 
 /** @brief Connect to the AWS IoT broker.
  *
- *  @details This function exposes the MQTT socket to main so that it can be
- *           polled on.
+ *  @details This function is synchronous and will only return success when the client has connected
+ *	     to the broker and all subscriptions (if any) have been acknowledged.
+ *	     If these conditions has not completed within CONFIG_AWS_IOT_CONNECT_TIMEOUT_SECONDS,
```

Comment:
has -> have

Reply:
Unsure, Isn't the subject in this case singlular? Client not clients.